patches/tModLoader/Terraria/GameContent/ShopHelper.cs.patch

```diff
@@ -182,14 +182,35 @@
  			if (aShoppingBiome.IsInBiome(player)) {
  				AddHappinessReportText("HateBiome", new {
  					BiomeName = BiomeNameByKey(aShoppingBiome.NameKey)
-@@ -341,7 +_,10 @@
+@@ -341,9 +_,31 @@
  	private bool IsNotReallyTownNPC(NPC npc)
  	{
  		int type = npc.type;
 +		/*
  		if (type == 37 || type == 368 || type == 453)
 +		*/
-+		if (type == 37 || type == 368 || NPCID.Sets.ActsLikeTownNPC[type])
++		if (NPCID.Sets.IsNotReallyTownNPC[type])
  			return true;
  
  		return false;
++	}
++
++	internal void RegisterTownNPCMoodLocalizations(ModNPC npc)
```

Comment:
I get that happiness and shopping are pretty intertwined, but I think this code is better off in `NPCLoader`